support-workers/src/main/scala/com/gu/emailservices/SupporterPlusEmailFields.scala

```diff
@@ -1,14 +1,17 @@
 package com.gu.emailservices
 
 import com.gu.emailservices.SubscriptionEmailFieldHelpers.{formatDate, hyphenate, mask}
+import com.gu.support.config.TouchPointEnvironment
 import com.gu.support.workers._
 import com.gu.support.workers.states.SendThankYouEmailState.SendThankYouEmailSupporterPlusState
 import org.joda.time.DateTime
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class SupporterPlusEmailFields(
+    paperFieldsGenerator: PaperFieldsGenerator,
     getMandate: String => Future[Option[String]],
```

Comment:
although the paperFieldsGenerator has getMandate already, it looks like we have a modified version copied into this class already, with different field names.  Perhaps we should split PaperFieldsGenerator into the payment realted stuff and the promo related stuff, so we don't have this loss of cohesion (but it's not affecting functionality, it just adds to code complexty.)

Reply:
I meant to say payment method related stuff, if you look in the class they are both there together, but S+ uses its own function for payment methods stuff (that is probably copied and modified from paperFieldsGenerator) in order to insert the payment stuff into the email.
Let's revisit this once this PR is out, as it does need a bit of naming at least as you mention.